core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -339,4 +349,12 @@ public boolean matchesIndices(String... indices) {
         }
         return false;
     }
+
+    public List<Object> handleTermsLookup(TermsLookup termsLookup) {
+        if (termsLookup.getIndex() == null) {
+            termsLookup.index(this.index().name());
+        }
+        termsLookup.index((termsLookup.getIndex() == null) ? this.index().name() : termsLookup.getIndex());
```

Comment:
I think there is some duplication here in the above lines. Anyways, as I said before I wouldn't mutate the provided terms lookup, I think this shuld happen in the caller and this tiny bit of logic should be moved to `toQuery` so it is clearer which index we query from the very beginning. Can you please make that change?